starter_code/elevator.js

```diff
@@ -1,19 +1,73 @@
+/*jshint esversion: 6 */
+
 class Elevator {
   constructor(){
     this.floor      = 0;
     this.MAXFLOOR   = 10;
+    this.waitingList = [];
+    this.passengers = [];
     this.requests   = [];
+    this.direction = "up";
+  }
+
+  start() {
+    let timer = setInterval(() => this.update(), 1000);
+    setTimeout(() => this.stop(timer), 10000);
+  }
+
+  stop(timer) {
```

Comment:
Better don't pass timer as a parameter of the function. You can use this.timer inside the function.